plugins/apps/threatbus_zmq_app/README.md

```diff
@@ -0,0 +1,167 @@
+
+### Subscribe at the Plugin
+
+To subscribe to Threat Bus via the zmq-app plugin, an app needs to send a JSON
+struct as follows to the `manage` endpoint of the plugin:
+
+```
+{
+  "action": "subscribe",
+  "topic": <TOPIC>,       # either 'threatbus/sighting' or 'threatbus/intel'
+  "snapshot": <SNAPSHOT>  # number of days for a snapshot, 0 for no snapshot
+}
+```
+In response, the app will either receive a `success` or `unsuccess` response.
+
+- Unsuccess response:
+  ```
+  {
+    "status": "unsuccess"
+  }
+  ```
+- Success response:
+  ```
+  {
+    "topic": <P2P_TOPIC>,
+    "pub_endpoint": 13371,
+    "sub_endpoint": 13372,
+    "status": "success",
+  }
+  ```
+
+  The `pub_endpoint` and `sub_endpoint` are the port numbers that are configured
+  in the Threat Bus plugin config. The app can access these endpoints following
+  the [ZeroMQ Pub/Sub](https://learning-0mq-with-pyzmq.readthedocs.io/en/latest/pyzmq/patterns/pubsub.html)
+  pattern. The plugin will publish messages on the `pub_endpoint` and accept
+  messages on the `sub_endpoint`.
+
+  The `topic` field of the response contains a unique topic for the client. That
+  topic _must_ be used to receive messages. The unique topic is a 32 characters
+  wide random string and guarantees that other subscribers won't accidentally
+  see traffic that should only be visible to the new client.
+  
+  For more details see below at `Pub/Sub via ZeroMQ`.
+
+### Unsubscribe from the Plugin
+
+To unsubscribe a connected app, it has to send a JSON struct to the `manage`
+endpoint of the plugin, as follows:
+
+```
+{
+  "action": "unsubscribe",
+  "topic": <P2P_TOPIC>       # the 32-characters random topic that the got during subscription handshake
+}
+```
+
+In response, the app will either receive a `success` or `unsuccess` response.
+
+- Unsuccess response:
+  ```
+  {
+    "status": "unsuccess"
+  }
+  ```
+- Success response:
+  ```
+  {
+    "status": "success"
+  }
+  ```
+
+### Pub/Sub via ZeroMQ
+
+Once an app has subscribed to Threat Bus via using the `manage` endpoint of the
+zmq-app plugin, it got a unique, random `p2p_topic` (see above). Threat Bus
+(the zmq-app) uses this topic to publish all kinds of messages to the plugin, be
+it `Intel`, `Sightings` or `SnapshotRequests`.
+
+ZeroMQ uses [prefix matching](https://zeromq.org/socket-api/#topics) for pub/sub
+connections. The zmq-app plugin leverages this feature to indicate the type of
+each sent message to the subscriber. Hence, app can simply match the topic
+suffix to determine the message type.
+
+For example, all `Intel` items will always be sent on the topic
+`p2p_topic + "intel"`, all `SnapshotRequest` items on the topic
+`p2p_topic + "snapshotrequest"` and so forth.
+
+### Snapshots
+
+Threat Bus' [snapshot](https://docs.tenzir.com/threatbus/features/snapshotting)
+feature allows apps to request a snapshot during subscription. But snapshots
+also should be provided by apps when Threat Bus requests it. This is done via
```

Comment:
"should" or will be provided? If this is a well-defined operation, it should be expressed with more rigor.

Reply:
Apps don't have to implement that feature, if they don't want to. E.g., the vast bridge does not implement it, but the MISP plugin does.

Reply:
I updated the text for clarity.